src/rust/engine/process_execution/src/remote.rs

```diff
@@ -134,19 +134,33 @@ impl CommandRunner {
                                 )
                               )
                             })
-                            .map(|operation| future::Loop::Continue(operation))
+                            // Reset `iter_num` on `MissingDigests`
+                            .map(|operation| future::Loop::Continue((operation, 0)))
                             .to_boxed()
                       },
                       ExecutionError::NotFinished(operation_name) => {
                         let mut operation_request =
                           bazel_protos::operations::GetOperationRequest::new();
                         operation_request.set_name(operation_name);
+
+                        let max_wait = 5000;
```

Comment:
This should be extracted into a constant.

Reply:
Done.